xbmc/Application.cpp

```diff
@@ -2502,8 +2503,60 @@ bool CApplication::PlayStack(CFileItem& item, bool bRestart)
   return PlayFile(selectedStackPart, "", true);
 }
 
+void CApplication::SelectMovieVersion(CFileItem& item)
```

Comment:
I don't think dumping even more GUI stuff into CApplication is the way to go unfortunately. We need some sort of mechanism to bridge this into GUI via messaging, I am not sure if we already have any pattern like this in the codebase.

Reply:
I don't disagree, though I'm not very familiar with this codebase and my C++ skills are mediocre at best. The only thing I can think of is maybe mimicking the resume/restart logic? But after doing some skimming it looks like maybe too much of that is stashed in the PlayerCore, and this definitely doesn't belong there either.